### PR TITLE
build: add ValveFileVDF library to cmake project

### PR DESCRIPTION
**Pull request by es3n1n**
Hi, first of all, thanks for this amazing library. 

This relatively small PR adds a declaration of `ValveFileVDF` library in CMakeLists.txt that could then be used easily in other codebases that are utilizing CMake in a bit more neat way:

```cmake
include(FetchContent)

FetchContent_Declare(ValveFileVDF SYSTEM
    GIT_REPOSITORY
    "https://github.com/TinyTinni/ValveFileVDF"
)
FetchContent_MakeAvailable(ValveFileVDF)
```

To make things a bit easier, I also moved the header to a separate `include` directory, which might be a somewhat breaking change for the library users, so I would be glad to hear any feedback on that, too. Other than that, I think everything is still the same way.

Please note that I didn't test the `fuzzing` project on my machine(it compiles just fine though) as it never got to work on Windows, but I guess this might be a skill issue.

### CODE REVIEW

File: tests/vdf_parser_test.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
 #include <string>
 
 #define TYTI_NO_L_UNDEF
-#include "../vdf_parser.hpp"
+#include "vdf_parser.hpp"
```
Comment (TinyTinni) [inline tests/vdf_parser_test.cpp:9]:
```suggestion
#include <vdf_parser.hpp>

```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,9 @@
 cmake_minimum_required (VERSION 3.6)
 project (vdf-Parser)
 
+add_library(ValveFileVDF INTERFACE)
+target_include_directories(ValveFileVDF INTERFACE "include")
```
Comment (TinyTinni) [inline CMakeLists.txt:5]: Can you change the target name to include a namespace and alias it?

I think exporting ValveFileVDF::ValveFileVDF is the standard and changing it at a later point would not be nice to people who use FetchContent, as they have to change the linking target again.

Perhaps another target like the python target might be added at a later point.

I would change target_include_directories change to HEADERS_SETS, but this is not important for this PR and can be done at a later point.

**Review by TinyTinni — CHANGES_REQUESTED** [review]
Hi,
thanks for the PR.

Adding a namespace to the target would be important to me in order to merge the change, it would be really nice if you could change it.

File: CMakeLists.txt (hunk 2 above)
Comment (es3n1n) [reply CMakeLists.txt:5]: I don't think that the target name can include a namespace by default, as when I am doing

```cmake
add_library(ValveFileVDF::ValveFileVDF INTERFACE)
```

CMake throws an error

```
  The target name "ValveFileVDF::ValveFileVDF" is reserved or not valid for
  certain CMake features, such as generator expressions, and may result in
  undefined behavior.
```

Although we can make the name `ValveFileVDF` and alias `ValveFileVDF::ValveFileVDF` to it, how does that sound?

Comment (TinyTinni) [reply CMakeLists.txt:5]: Thanks, thats sounds fine for me.

**Comment by TinyTinni** [comment]
Pipeline fails. I think cmake/the compiler recognizes the header file now as external.

Not sure if it works when you disable the C++17 deprecation warnings as a PRIVATE compile_definitions to the library.
```
target_compile_definitions(ValveFileVDF PRIVATE "-D_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS")
```
This should silence the warning only in the header, or?

If it doesn't work, it might be a good idea to just remove the following line:
```
set_property(TARGET tests PROPERTY COMPILE_WARNING_AS_ERROR ON)
```

**Review by TinyTinni — APPROVED** [review]

**Comment by es3n1n** [comment]
Hmm, I think I found a workaround for that exact error https://github.com/es3n1n/ValveFileVDF/commit/8b4c6b87666c93939c5b96727ef6674690b5bb33

But the thing is that we can't add a PRIVATE compile definition for an interface, thus we have to set it as an INTERFACE and it then will be added to the targets that are using the ValveFileVDF library. This doesn't sound too bad honestly, but I believe that some library users wouldn't be too happy to have the `_LIBCPP_DISABLE_DEPRECATION_WARNINGS` definition in their codebase.

Alternatively, we could probably disable them only in our tests/fuzzing projects.

While investigating I referenced the following libcpp config, it may or may not differ on different libcpp versions: https://github.com/plentico/osxcross-target/blob/master/SDK/MacOSX11.3.sdk/usr/include/c%2B%2B/v1/__config#L1032

**Comment by TinyTinni** [comment]
Adding the flag `_LIBCPP_DISABLE_DEPRECATION_WARNINGS` to the already existing c++17 warning disable the warnings in the tests/fuzzer sounds like a good idea.

**Comment by es3n1n** [comment]
Alright, the macos pipeline seems to be passing now.

**Comment by TinyTinni** [comment]
Thank you!

**Pull request merged**